zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithSuspensionInstructionTest.java

```diff
@@ -0,0 +1,505 @@
+import io.camunda.zeebe.protocol.record.intent.IncidentIntent;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceIntent;
+import io.camunda.zeebe.protocol.record.intent.ProcessMessageSubscriptionIntent;
+import io.camunda.zeebe.protocol.record.intent.UserTaskIntent;
+import io.camunda.zeebe.test.util.BrokerClassRuleHelper;
+import io.camunda.zeebe.test.util.record.RecordingExporter;
+import io.camunda.zeebe.test.util.record.RecordingExporterTestWatcher;
+import org.assertj.core.groups.Tuple;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class CreateProcessInstanceWithSuspensionInstructionTest {
+
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Rule public final BrokerClassRuleHelper helper = new BrokerClassRuleHelper();
+
+  @Test
+  public void shouldSuspendProcessInstanceWhenElementIsCompleted() {
+    // given
+    final String processId = "process";
+    final String elementBeforeSuspension = "element";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .manualTask(elementBeforeSuspension)
+                .endEvent()
+                .done())
+        .deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeSuspendInstruction(elementBeforeSuspension)
+            .create();
+
+    // then
+    final var result =
+        RecordingExporter.processInstanceRecords()
+            .onlyEvents()
+            .withProcessInstanceKey(processInstanceKey)
+            .limit(processId, ProcessInstanceIntent.ELEMENT_SUSPENDED)
+            .filter(
+                record ->
+                    record.getValue().getElementId().equals(processId)
+                        || record.getValue().getElementId().equals(elementBeforeSuspension));
+
+    assertThat(result)
+        .extracting(Record::getIntent, record -> record.getValue().getElementId())
+        .containsSequence(
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_COMPLETED, elementBeforeSuspension),
+            Tuple.tuple(ProcessInstanceIntent.ELEMENT_SUSPENDED, processId));
+
+    final var processInstanceRecord =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_SUSPENDED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    ENGINE.writeRecords(
+        RecordToWrite.command()
+            .key(processInstanceKey)
+            .processInstance(
+                ProcessInstanceIntent.COMPLETE_ELEMENT, processInstanceRecord.getValue()));
```

Comment:
🔧 This is not necessary. The Manual Task is activated and completed automatically by the engine, i.e. it does not have a wait state and does not require external interaction.

Reply:
I might be misunderstanding your point here, but this record doesn't complete a Manual Task. Instead, it aims to verify that after process is suspended, new records aiming to move it to another state (e.g. completed) will be rejected.